packages/stream/solana/distributor/client.ts

```diff
@@ -0,0 +1,268 @@
+import BN from "bn.js";
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  NATIVE_MINT,
+  TOKEN_PROGRAM_ID,
+  createTransferCheckedInstruction,
+  getMint,
+} from "@solana/spl-token";
+import {
+  Connection,
+  PublicKey,
+  SystemProgram,
+  TransactionInstruction,
+  Transaction,
+  Commitment,
+  ConnectionConfig,
+} from "@solana/web3.js";
+
+import { DISTRIBUTOR_PROGRAM_ID } from "./constants";
+import { ICluster, ITransactionResult } from "../../common/types";
+import {
+  IClaimData,
+  IClawbackData,
+  ICreateDistributorData,
+  ICreateDistributorResult,
+  IGetClaimsData,
+  IGetDistributors,
+} from "./types";
+import { ICreateSolanaExt, IInteractStreamSolanaExt } from "../types";
+import { ata, checkOrCreateAtaBatch } from "../utils";
+import {
+  ClaimLockedAccounts,
+  ClawbackAccounts,
+  NewClaimAccounts,
+  NewClaimArgs,
+  NewDistributorAccounts,
+  NewDistributorArgs,
+  claimLocked,
+  clawback,
+  newClaim,
+  newDistributor,
+} from "./generated/instructions";
+import { prepareWrappedAccount } from "../instructions";
+import { ClaimStatus, MerkleDistributor } from "./generated/accounts";
+import { getClaimantStatusPda, getDistributorPda, wrappedSignAndExecuteTransaction } from "./utils";
+
+interface IInitOptions {
+  clusterUrl: string;
+  cluster?: ICluster;
+  commitment?: Commitment | ConnectionConfig;
+  programId?: string;
+}
+
+export default class SolanaDistributorClient {
+  private connection: Connection;
+
+  private programId: PublicKey;
+
+  private commitment: Commitment | ConnectionConfig;
+
+  /**
+   * Create Stream instance
+   */
+  constructor({
+    clusterUrl,
+    cluster = ICluster.Mainnet,
```

Comment:
When I touched our js-sdk api for the first time I had an issue with this setup. 
As `clusterUrl` and `cluster` are separated, it wasn't obvious for me that I have to pass them both when I switch to ``devnet` for instance. Probably we should combine them to an one parameter?

like 
```
type ClusterConnectionInput = {
  clusterUrl: string,
  cluster: <some predefined union type>
}
```